### PR TITLE
Remove unnecessary code on UWP

### PR DESCRIPTION
**Pull request by Jamminroot**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR will remove redundant code from cake script.
Only to be merged with other related PRs.

[AB#77177](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/77177)

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -51,7 +51,6 @@ var ExternalUnityPackages = new [] {
 var UwpIL2CPPDependencies = new [] {
     new NugetDependency("sqlite-net-pcl", "1.3.1", "UAP, Version=v10.0")
 };
-var UwpIL2CPPJsonUrl = SdkStorageUrl + "Newtonsoft.Json.dll";
```
Comment (MatkovIvan) [inline build.cake]: It should be downloaded from Nuget instead

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/AppCenterInternal.cs [hunk 2]
```diff
@@ -199,27 +199,9 @@ private static void Prepare()
...
-                     * IL2CPP does not yet support calling this projected method.
-                     */
                     UnityApplicationSettings.Initialize();
                     UWPAppCenter.SetApplicationSettingsFactory(new UnityApplicationSettingsFactory());
-
-                    /**
-                     * Workaround for another IL2CPP issue.
-                     * System.Net.Http.HttpClient doesn't work properly so, replace to unity specific implementation.
-                     */
                     UWPAppCenter.SetChannelGroupFactory(new UnityChannelGroupFactory());
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/AppCenterInternal.cs]: Why only comment is deleted? I guess this workarounds can be removed entirely.

File: build.cake [hunk 3]
```diff
@@ -49,9 +49,9 @@ var ExternalUnityPackages = new [] {
 
 // UWP IL2CPP dependencies.
 var UwpIL2CPPDependencies = new [] {
-    new NugetDependency("sqlite-net-pcl", "1.3.1", "UAP, Version=v10.0")
+    new NugetDependency("sqlite-net-pcl", "1.3.1", "UAP, Version=v10.0"),
```
Comment (MatkovIvan) [inline build.cake]: Resolve conflicts with base branch

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/AppCenterInternal.cs [hunk 4]
```diff
@@ -200,26 +205,17 @@ private static void Prepare()
...
-#if ENABLE_IL2CPP
-#pragma warning disable 612
                     /**
                      * Workaround for known IL2CPP issue.
                      * See https://issuetracker.unity3d.com/issues/il2cpp-use-of-windows-dot-foundation-dot-collections-dot-propertyset-throws-a-notsupportedexception-on-uwp
                      *
                      * NotSupportedException: Cannot call method
                      * 'System.Boolean System.Runtime.InteropServices.WindowsRuntime.IMapToIDictionaryAdapter`2::System.Collections.Generic.IDictionary`2.TryGetValue(TKey,TValue&)'.
                      * IL2CPP does not yet support calling this projected method.
+                     * TODO: Can be removed after migration for settings is added.
```
Comment (guperrot) [inline Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/AppCenterInternal.cs:217]: What is that?

Comment (MatkovIvan) [reply Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/AppCenterInternal.cs:217]: This workaround is not required anymore but it uses a different place to store settings (enabling state etc). So to remove it we should provide some migration. So we can just keep it for now

Comment (Jamminroot) [reply Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/AppCenterInternal.cs:217]: Clarified that in comment

File: Assets/AppCenter/Editor/AppCenterPostBuild.cs [hunk 5]
```diff
@@ -36,12 +30,10 @@ public class AppCenterPostBuild : IPostprocessBuild
     private const string AppIl2cppXaml = "App.xaml.cpp";
```
Comment (MatkovIvan) [inline Assets/AppCenter/Editor/AppCenterPostBuild.cs:28]: Remove 2 lines above

**Review by MatkovIvan — CHANGES_REQUESTED** [review]
Blocking until tested with new dlls

File: Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/AppCenterInternal.cs [hunk 6]
```diff
@@ -207,6 +207,8 @@ private static void Prepare()
 
                     /**
                      * Workaround for known IL2CPP issue.
+                     * Currently it is not required, but since this workaround stores settings in separate file, 
+                     * and not generic unity mechanism, a migration is required to transition users who store data in old file to built-in mechanism.
```
Comment (guperrot) [inline Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/AppCenterInternal.cs:211]: I don't see the todo in the feature work items by the way, are we tracking that somewhere?

Comment (MatkovIvan) [reply Assets/AppCenter/Plugins/AppCenterSDK/Core/UWP/AppCenterInternal.cs:211]: It was agreed to keep this without changes

**Review by MatkovIvan — APPROVED** [review]

**Review by olmisa — APPROVED** [review]

File: Assets/AppCenter/Plugins/iOS/Analytics/AppCenterAnalytics.framework.meta [hunk 7]
```diff
@@ -1,141 +0,0 @@
-fileFormatVersion: 2
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/iOS/Analytics/AppCenterAnalytics.framework.meta]: Wrong change. Must be reverted

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**